libs/ledger-live-common/src/families/celo/api/hubble.ts

```diff
@@ -89,5 +135,40 @@ export const getAccountDetails = async (address: string, accountId: string) => {
     balance,
     spendableBalance,
     operations,
+    lockedBalance,
+    nonvotingLockedBalance,
   };
 };
+
+export const getValidatorGroups = async (): Promise<CeloValidatorGroup[]> => {
+  const validatorGroups = await fetchValidatorGroups();
+
+  const result = validatorGroups.map((validatorGroup) => ({
+    //TODO: toLowerCase()?
+    address: validatorGroup.address,
+    name: validatorGroup.name || validatorGroup.address,
+    votes: new BigNumber(validatorGroup.active_votes).plus(
+      new BigNumber(validatorGroup.pending_votes)
+    ),
+  }));
+  return customValidatorGroupsOrder(result);
+};
+
+const customValidatorGroupsOrder = (validatorGroups): CeloValidatorGroup[] => {
```

Comment:
There's a good amount of mutation happening here, which I personally try to stay away from.  A way to refactor this to avoid mutation would be something like:

```
const defaultValidatorGroup = validatorGroups.find(
    isDefaultValidatorGroup
  );

const sortedValidatorGroups = [...validatorGroups].sort((a, b) =>
    b.votes.minus(a.votes)
  ).filter(group => {
  return !defaultValidatorGroup || !isDefaultValidatorGroup(group);
});

return [
  defaultValidatorGroup,
  ...sortedValidatorGroups
];
```

Reply:
good idea, thanks! 👍